src/Handlers/AbstractPaymentMethodHandler.php

```diff
@@ -434,7 +438,12 @@ protected function preparePaymentsRequest(
 
         if (!empty($request['storePaymentMethod']) && $request['storePaymentMethod'] === true) {
             $request['recurringProcessingModel'] = 'CardOnFile';
-            $request['shopperInteraction'] = 'Ecommerce';
+        }
+
+        if ($paymentMethodEntity->getHandlerIdentifier() === OneClickPaymentMethodHandler::class) {
```

Comment:
this method will be called from the payment method handler itself so it should be enough to to the following here
```suggestion
        if (static::getPaymentMethodCode() === 'storedPaymentMethods') {
```

Reply:
or
```suggestion
        if (static::class === OneClickPaymentMethodHandler::class) {
```

Reply:
Thank you for the review @peterojo.